vscode/src/diffView/fileModel.ts

```diff
@@ -0,0 +1,222 @@
+      const loc =
+        item instanceof vscode.Location
+          ? item
+          : new vscode.Location(item.targetUri, item.targetRange);
+
+      if (!last || KonveyorFileModel._compareUriIgnoreFragment(last.uri, loc.uri) !== 0) {
+        last = new FileItem(loc.uri.with({ fragment: "" }), "", this);
+        this.items.push(last);
+      }
+    }
+  }
+
+  private static _compareUriIgnoreFragment(a: vscode.Uri, b: vscode.Uri): number {
+    const aStr = a.with({ fragment: "" }).toString();
+    const bStr = b.with({ fragment: "" }).toString();
+    if (aStr < bStr) {
+      return -1;
+    } else if (aStr > bStr) {
+      return 1;
+    }
+    return 0;
+  }
+
+  private static _compareLocations(
+    a: vscode.Location | vscode.LocationLink,
+    b: vscode.Location | vscode.LocationLink,
+  ): number {
+    const aUri = a instanceof vscode.Location ? a.uri : a.targetUri;
+    const bUri = b instanceof vscode.Location ? b.uri : b.targetUri;
+    if (aUri.toString() < bUri.toString()) {
+      return -1;
+    } else if (aUri.toString() > bUri.toString()) {
+      return 1;
+    }
+
+    const aRange = a instanceof vscode.Location ? a.range : a.targetRange;
+    const bRange = b instanceof vscode.Location ? b.range : b.targetRange;
+    if (aRange.start.isBefore(bRange.start)) {
+      return -1;
+    } else if (aRange.start.isAfter(bRange.start)) {
+      return 1;
+    } else {
+      return 0;
+    }
+  }
+
+  // --- adapter
+
+  get message() {
+    if (this.items.length === 0) {
+      return "No results.";
+    }
+    const filesCount = this.items.length;
+    return `Edits in ${filesCount} file(s)`;
+  }
+
+  location(item: FileItem) {
+    return new vscode.Location(item.uri, new vscode.Position(0, 0));
+  }
+
+  next(item: FileItem): FileItem {
+    return this._move(item, true) ?? item;
+  }
+
+  previous(item: FileItem): FileItem {
+    return this._move(item, false) ?? item;
+  }
+
+  private _move(item: FileItem, fwd: boolean): FileItem | void {
+    const delta = fwd ? +1 : -1;
+
+    const _move = (item: FileItem): FileItem => {
+      const idx = (this.items.indexOf(item) + delta + this.items.length) % this.items.length;
+      return this.items[idx];
+    };
+
+    return _move(item);
+  }
+
+  remove(item: FileItem) {
+    const idx = this.items.indexOf(item);
+    if (idx >= 0) {
+      this.items.splice(idx, 1);
+    }
+    this._onDidChange.fire(undefined);
+  }
+
+  apply(item: FileItem) {
+    vscode.window.showInformationMessage(`[TODO] Apply resolutions to ${item.uri.fsPath}`);
+  }
+
+  revert(item: FileItem) {
+    vscode.window.showInformationMessage(
+      `[TODO] Discard local changes to resolutions for ${item.uri.fsPath}`,
+    );
+  }
+
+  async asCopyText() {
+    let result = "";
+    for (const item of this.items) {
+      result += `${await item.asCopyText()}\n`;
+    }
+    return result;
+  }
+}
+
+export class KonveyorTreeDataProvider implements vscode.TreeDataProvider<FileItem> {
```

Comment:
I reject the idea that we need a treedataprovider at this stage, the navigation of problems to solve and the files changed by a particular solution from kai/rpc-server should be in the webview.